src/e2e-test/java/io/cdap/plugin/servicenowsink/actions/ServiceNowSinkPropertiesPageActions.java

```diff
@@ -53,11 +62,93 @@ public static void verifyIfRecordExistsInServiecNowReceivingSlipLineTable(String
+    logger.info("Response of ServiceNow : " + responseFromServiceNowTable);
+    Assert.assertTrue(compareValueOfBothResponses(responseFromServiceNowTable, bigQueryResponseInMap));
+  }
+
+  public static void verifyIfRecordUpdatedInServiceNowIsCorrect(String query, String tableName)
+    throws IOException, InterruptedException, OAuthProblemException, OAuthSystemException {
+
+    getRecordFromServiceNowTable(query, tableName);
+    TableResult bigQueryTableData = getBigQueryTableData(TestSetupHooks.bqSourceDataset, TestSetupHooks.bqSourceTable);
+    String bigQueryJsonResponse = bigQueryTableData.getValues().iterator().next().get(0).getValue().toString();
+    Gson gson = new Gson();
+    JsonObject jsonObject = gson.fromJson(bigQueryJsonResponse, JsonObject.class);
+    Map<String, Object> bigQueryResponseInMap = gson.fromJson(jsonObject.toString(), Map.class);
+
+    logger.info("Response of BigQuery : " + bigQueryResponseInMap);
+    logger.info("Response of ServiceNow : " + responseFromServiceNowTable);
+    Assert.assertTrue(compareValueOfBothResponses(responseFromServiceNowTable, bigQueryResponseInMap));
+  }
+
+  public static TableResult getBigQueryTableData(String dataset, String table)
+    throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String selectQuery = "SELECT TO_JSON(t) result FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static boolean compareValueOfBothResponses(Map<String, String> serviceNowResponseMap,
+                                                    Map<String, Object> bigQueryResponseMap) {
+    if (serviceNowResponseMap.isEmpty() || bigQueryResponseMap.isEmpty()) {
+      return false;
+    }
+    boolean result = false;
+    Set<String> bigQueryKeySet = bigQueryResponseMap.keySet();
+
+    for (String setValue : bigQueryKeySet) {
+      Object serviceNowValue = serviceNowResponseMap.get(setValue);
+      Object bigQueryValue = bigQueryResponseMap.get(setValue);
+
+      if (bigQueryValue instanceof Double) {
+        String bigDecimalValue = new BigDecimal(String.valueOf(bigQueryValue)).setScale(DEFAULT_SCALE,
+                                                                               RoundingMode.HALF_UP).toString();
+        result = serviceNowValue.equals(bigDecimalValue);
+      } else if (checkBigQueryDateFormat(bigQueryValue.toString()) != null) {
+        SimpleDateFormat serviceNowDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        String bigQueryFormattedValue = serviceNowDateFormat.format(checkBigQueryDateFormat(bigQueryValue.toString()));
+        result = String.valueOf(serviceNowValue).equals(bigQueryFormattedValue);
+      } else {
+        result = String.valueOf(serviceNowValue).equals(String.valueOf(bigQueryValue));
+      }
+    }
+    return result;
+  }
 
+  public static Date checkBigQueryDateFormat(String value) {
```

Comment:
make this method private and add description on top

Reply:
Done